libs/blocks/global-navigation/global-navigation.js

```diff
@@ -152,77 +149,56 @@ class Gnav {
 
   decorateProfile = async () => {
     const { blockEl, decoratedEl } = this.blocks.profile;
-    if (!blockEl) return null;
+    if (!blockEl) return;
 
     const accessToken = window.adobeIMS.getAccessToken();
     const { env } = getConfig();
-    const profileRes = accessToken
+    this.blocks.profile.profileRes = accessToken
       ? await fetch(`https://${env.adobeIO}/profile`, { headers: new Headers({ Authorization: `Bearer ${accessToken.token}` }) })
       : {};
-    if (profileRes.status !== 200) return this.decorateSignIn();
-
-    if (blockEl.children.length > 1) decoratedEl.classList.add('has-menu');
-    decoratedEl.closest('nav.gnav')?.classList.add('signed-in');
-    const { sections, user: { avatar } } = await profileRes.json();
-    const avatarImgEl = toFragment`<img class="feds-profile-img" src="${avatar}"></img>`;
-    const profileButtonEl = toFragment`
-        <button 
-          class="feds-profile-button" 
-          aria-expanded="false" 
-          aria-controls="feds-profile-menu"
-          aria-label="Profile button"
-          daa-ll="Account"
-          aria-haspopup="true"
-        > 
-          ${avatarImgEl}
-        </button>
-      `;
-    this.decorateProfileMenu({ avatarImgEl, sections, profileRes, profileButtonEl });
-    decoratedEl.append(profileButtonEl);
-    return null;
-  };
-
-  decorateProfileMenu = ({ avatarImgEl, sections, profileRes, profileButtonEl }) => {
-    const { blockEl, decoratedEl } = this.blocks.profile;
-    let decorating;
-    const decorate = async (event) => {
-      if (decorating) return;
-      decorating = true;
-      await this.loadDelayed();
-
-      // Forward click on the profile button it was clicked. Otherwise users need to click twice
-      if (event) {
-        decoratedEl.addEventListener('feds:events:profileReady', () => decoratedEl.click(), { once: true });
-      }
 
-      // TODO integrate placeholders here
-      this.blocks.profile.instance = new this.Profile({
-        accountLinkEl: blockEl.querySelector('div > div > p:nth-child(2) a'), // TODO placeholders
-        manageTeamsEl: blockEl.querySelector('div > div > p:nth-child(3) a'), // TODO placeholders
-        manageEnterpriseEl: blockEl.querySelector('div > div > p:nth-child(4) a'), // TODO placeholders
-        signOutEl: blockEl.querySelector('div > div > p:nth-child(5) a'), // TODO placeholders
-        localMenu: blockEl.querySelector('h5')?.parentElement,
-        toggleMenu: this.menuControls.toggleMenu,
-        decoratedEl,
-        avatarImgEl,
-        sections,
-        profileRes,
-        profileButtonEl,
+    if (this.blocks.profile.profileRes.status !== 200) {
+      const [decorateSignIn] = await Promise.all([
+        loadBlock('./blocks/profile/signIn.js'),
+        loadStyles('./blocks/profile/signIn.css'),
+      ]);
+      const signInEl = await decorateSignIn({ blockEl, decoratedEl, toFragment, getPlaceholder });
```

Comment:
What's the reasoning behind sending utility methods as params rather than loading them where needed?

Reply:
We should revert and spend some time investigating how to mock

Reply:
Properly set this up now